cdap-ui/app/cdap/components/PipelineContextMenu/index.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React from 'react';
+import { ContextMenu, IContextMenuOption } from 'components/ContextMenu';
+import WranglerConnection from 'components/PipelineContextMenu/WranglerConnection';
+import If from 'components/If';
+import PropTypes from 'prop-types';
+import { CopyFromClipBoard } from 'services/Clipboard';
+import { objectQuery } from 'services/helpers';
+import { INode, IConnection } from 'components/PipelineContextMenu/PipelineTypes';
+import { INewWranglerConnection } from 'components/PipelineContextMenu/WranglerConnection';
+import { GLOBALS } from 'services/global-constants';
+
+export interface IStage {
+  nodes: INode[];
+  connections: IConnection[];
+}
+interface IPipelineContextMenuProps {
+  onNodesPaste: (stages: IStage) => void;
+  onWranglerSourceAdd: INewWranglerConnection;
+  pipelineArtifactType: 'cdap-data-pipeline' | 'cdap-data-streams';
+  onZoomIn: () => void;
+  onZoomOut: () => void;
+  fitToScreen: () => void;
+  prettyPrintGraph: () => void;
+}
+
+async function getNodesFromClipBoard(): Promise<IStage | undefined> {
+  let clipText;
+  try {
+    clipText = await CopyFromClipBoard();
+  } catch (e) {
+    return Promise.reject();
+  }
+  return Promise.resolve(getClipboardData(clipText));
+}
+
+function getClipboardData(text): IStage | undefined {
```

Comment:
these code seemed similar to https://github.com/cdapio/cdap/pull/11773?

Reply:
The one in #11773 is to add a generic function that will use browser APIs to copy to/from clipboard. These functions actually use those and get data and parse and has logic that is specific to pipeline in this case.

Reply:
Have changed the name of this function to `parseClipboardData` to be make the distinction. Let me know if it makes sense.